x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java

```diff
@@ -59,36 +71,124 @@ public DataFrameTransformPersistentTasksExecutor(Client client,
+            }
+        );
+
+        // <1> Validate the transform, assigning it to the indexer, and get the field mappings
+        ActionListener<DataFrameTransformConfig> getTransformConfigListener = ActionListener.wrap(
+            config -> {
+                if (config.isValid()) {
+                    indexerBuilder.setTransformConfig(config);
+                    SchemaUtil.getDestinationFieldMappings(client, config.getDestination().getIndex(), getFieldMappingsListener);
+                } else {
+                    markAsFailed(buildTask,
+                        DataFrameMessages.getMessage(DataFrameMessages.DATA_FRAME_TRANSFORM_CONFIGURATION_INVALID, transformId));
+                }
+            },
+            error -> {
+                String msg = DataFrameMessages.getMessage(DataFrameMessages.FAILED_TO_LOAD_TRANSFORM_CONFIGURATION, transformId);
+                logger.error(msg, error);
+                markAsFailed(buildTask, msg);
             }
-        ));
+        );
+        // <0> Get the transform config
+        transformsConfigManager.getTransformConfiguration(transformId, getTransformConfigListener);
     }
 
-    private void scheduleTask(DataFrameTransformTask buildTask, SchedulerEngine.Job schedulerJob, String id) {
+    private void markAsFailed(DataFrameTransformTask task, String reason) {
+        CountDownLatch latch = new CountDownLatch(1);
+
+        task.markAsFailed(reason, new LatchedActionListener<>(ActionListener.wrap(
+            nil -> {},
+            failure -> logger.error("Failed to set task [" + task.getTransformId() +"] to failed", failure)
+        ), latch));
+        try {
+            latch.await(90, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            logger.error("timeout waiting for task [" + task.getTransformId() + "] to be marked as failed in cluster state", e);
+        }
+    }
+
+    private void scheduleAndStartTask(DataFrameTransformTask buildTask,
+                                      SchedulerEngine.Job schedulerJob,
+                                      String id,
+                                      ActionListener<StartDataFrameTransformTaskAction.Response> listener) {
         // Note that while the task is added to the scheduler here, the internal state will prevent
         // it from doing any work until the task is "started" via the StartTransform api
         schedulerEngine.register(buildTask);
         schedulerEngine.add(schedulerJob);
-
         logger.info("Data frame transform [" + id + "] created.");
+        // If we are stopped, and it is an initial run, this means we have never been started,
+        // attempt to start the task
+        if (buildTask.getState().getTaskState().equals(DataFrameTransformTaskState.STOPPED) && buildTask.isInitialRun()) {
+            buildTask.start(listener);
+        } else {
+            logger.info("No need to start task. Its current state is: " + buildTask.getState().getIndexerState());
```

Comment:
I think this and likely the one above seem leftovers from the early days, should we reduce the loglevel to debug?

Reply:
These log messages should occur few enough times that they won't really spam the output. It will also be beneficial to have them in the future if this change does break when we test in multi-node. 

I can change to debug if you feel strongly about it though.

Reply:
debug would be better. We should review the log messages in general once the code churn has settled down